### PR TITLE
[Merged by Bors] - feat: add tech debts variation in PR summary

### PR DESCRIPTION
**Pull request by adomani**
See #18943 for a sample output of the script.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: scripts/technical-debt-metrics.sh [hunk 1]
```diff
@@ -25,10 +25,27 @@ IFS=$'\n\t'
...
+  *)
+    currCommit="${1:-"$(git rev-parse HEAD)"}"
+    refCommit="${2:-"$(git log --pretty=%H --since="$(date -I -d 'last week')" | tail -n -1)"}"
+    >&2 printf '***  NO pr_summary passed  ***\n'
+    ;;
+esac
+
+###currCommit="${1:-"$(git rev-parse HEAD)"}"
+#### Similarly for the second argument.
+###refCommit="${2:-"$(git log --pretty=%H --since="$(date -I -d 'last week')" | tail -n -1)"}"
```
Comment (jcommelin) [inline scripts/technical-debt-metrics.sh]: Are these intended to stay?

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

File: .github/workflows/PR_summary.yml [hunk 2]
```diff
@@ -95,6 +95,9 @@ jobs:
         currentHash="$(git rev-parse HEAD)"
         hashURL="https://github.com/${{ github.repository }}/pull/${{ github.event.pull_request.number }}/commits/${currentHash}"
 
-        message="$(printf '%s [%s](%s)%s\n\n%s\n\n---\n\n%s\n' "${title}" "$(git rev-parse --short HEAD)" "${hashURL}" "${high_percentages}" "${importCount}" "${declDiff}")"
+        ## Technical debts variation
```
Comment (grunweg) [inline .github/workflows/PR_summary.yml:98]:
```suggestion
        ## Technical debt changes

File: scripts/technical-debt-metrics.sh [hunk 3]
```diff
@@ -98,6 +111,8 @@ printf '%s|%s\n' "$(printf '%s' "${deprecatedFiles}" | wc -l)" "\`Deprecated\` f
 printf '%s|%s\n\n' "$(printf '%s\n' "${deprecatedFiles}" | grep total | sed 's= total==')"  'total LoC in `Deprecated` files'
 }
 
+report () {
+
 # collect the technical debts and the line counts of the deprecated file from the current mathlib
```
Comment (grunweg) [inline scripts/technical-debt-metrics.sh:116]:
```suggestion
# Collect the technical debt metrics and the line counts of all deprecated files from current mathlib.

File: scripts/technical-debt-metrics.sh [hunk 4]
```diff
@@ -125,6 +140,21 @@ printf $'```spoiler Changed \'Deprecated\' lines by file\n%s\n```\n' "$(
...
 printf '\nCurrent commit [%s](%s)\n' "${currCommit:0:10}" "${baseURL}/${currCommit}"
 printf 'Reference commit [%s](%s)\n' "${refCommit:0:10}"  "${baseURL}/${refCommit}"
+}
+
+if [ "${1:-}" == "pr_summary" ]
+then
+  rep="$(report | awk -F'|' 'BEGIN{backTicks=0} /^```/{backTicks++} ((!/^```/) && (backTicks % 2 == 0) && !($3 == "0")) {print $0}')"
+  if [ "$(wc -l <<<"${rep}")" -le 5 ]
+  then
+    printf '<details><summary>No changes to technical debts.</summary>\n'
```
Comment (grunweg) [inline scripts/technical-debt-metrics.sh:150]:
```suggestion
   printf '<details><summary>No changes to technical debt.</summary>\n'

File: scripts/technical-debt-metrics.sh [hunk 5]
```diff
@@ -125,6 +140,21 @@ printf $'```spoiler Changed \'Deprecated\' lines by file\n%s\n```\n' "$(
...
+  else
+    printf '<details><summary>Changes to technical debts</summary>\n\n%s\n' "${rep}"
```
Comment (grunweg) [inline scripts/technical-debt-metrics.sh:152]:
```suggestion
    printf '<details><summary>Changes to technical debt</summary>\n\n%s\n' "${rep}"

**Review by grunweg — COMMENTED** [review]
Thanks for doing this - this is really useful! (Also for PRs adding `erw`, for example.)
My inner grammar nerd complains, that "debt" is not countable. What do you think about the following editorial tweaks?

**Review by adomani — COMMENTED** [review]
Michael, thank you for the suggestions!

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**